api/taxonomies/utils.py

```diff
@@ -1,7 +1,11 @@
-from django.db.models import Count
+from django.db.models import BooleanField, Case, Count, When
 
 def optimize_subject_query(subject_queryset):
     """
     Optimize subject queryset for TaxonomySerializer
     """
-    return subject_queryset.prefetch_related('parent', 'provider').annotate(children_count=Count('children'))
+    return subject_queryset.prefetch_related('parent', 'provider').annotate(children_count=Count('children'), is_other=Case(
+        When(text__startswith='Other', then=True),
```

Comment:
Might consider setting is_other `Case` in its own var, but that's just a minor readability change.

Reply:
I'm unsure whether chaining `annotate` calls actually affects performance, compared to multiple annotations in the same call, but I imagine it could. I can make this more readable, though.